youtube_dl/extractor/gaskrank.py

```diff
@@ -89,35 +82,25 @@ def fix_json(code):
         if average_rating:
             average_rating = float_or_none(average_rating.replace(',', '.'))
 
-        playlist = self._parse_json(
-            self._search_regex(
-                r'playlist\s*:\s*\[([^\]]*)\]',
-                webpage, 'playlist', default='{}'),
-            display_id, transform_source=fix_json, fatal=False)
-
         video_id = self._search_regex(
             r'https?://movies\.gaskrank\.tv/([^-]*?)(-[^\.]*)?\.mp4',
-            playlist.get('0').get('src'), 'video id')
+            webpage, 'video id')
 
-        formats = []
-        for key in playlist:
-            formats.append({
-                'url': playlist[key]['src'],
-                'format_id': key,
-                'quality': playlist[key].get('quality')})
-        self._sort_formats(formats, field_preference=['format_id'])
+        entries = self._parse_html5_media_entries(url, webpage, video_id)
+        if entries:
+            for entry in entries:
+                entry.update({
+                    'id': video_id,
+                    'title': title,
+                    'categories': categories,
+                    'display_id': display_id,
+                    'uploader_id': uploader_id,
+                    'upload_date': upload_date,
+                    'uploader_url': uploader_url,
+                    'tags': tags,
+                    'view_count': view_count,
+                    'average_rating': average_rating,
+                })
+                self._sort_formats(entry['formats'])
 
-        return {
-            'id': video_id,
-            'title': title,
-            'formats': formats,
-            'thumbnail': thumbnail,
-            'categories': categories,
-            'display_id': display_id,
-            'uploader_id': uploader_id,
-            'upload_date': upload_date,
-            'uploader_url': uploader_url,
-            'tags': tags,
-            'view_count': view_count,
-            'average_rating': average_rating,
-        }
+        return self.playlist_result(entries)
```

Comment:
There is always only one video on the page thus you should take first entry and return as regular info dict.